src/gd_kotlin.cpp

```diff
@@ -330,7 +344,12 @@ void GDKotlin::unregister_classes(jni::Env& p_env, jni::JObjectArray p_classes)
 }
 
 KtClass* GDKotlin::find_class(const String& p_script_path) {
-    StringName class_name = p_script_path.trim_prefix(scripts_root).trim_suffix(".kt").replace("/", ".");
+    String class_name_as_string = p_script_path.trim_prefix("res://").trim_suffix(".kt");
```

Comment:
This should be merged after `KtClasse` are cached in `KotlinScript`.
If we cannot cache them we'll have to find another way to retreive `KtClass`
Those lines will cost too much.

Reply:
Indeed. I'll fix the rest as well once we have the caching merged